ecpy_hqc_legacy/tasks/tasks/instr/synthHD_tasks.py

```diff
@@ -0,0 +1,458 @@
+# -*- coding: utf-8 -*-
+# -----------------------------------------------------------------------------
+# Copyright 2015-2016 by EcpyHqcLegacy Authors, see AUTHORS for more details.
+#
+# Distributed under the terms of the BSD license.
+#
+# The full license is in the file LICENCE, distributed with this software.
+# -----------------------------------------------------------------------------
+"""Task perform measurements with a synthHD.
+
+"""
+from __future__ import (division, unicode_literals, print_function,
+                        absolute_import)
+
+import time
+import re
+import numbers
+from inspect import cleandoc
+
+import numpy as np
+from atom.api import (Unicode, Int, Bool, Enum, set_default,
+                      Value, List)
+
+from ecpy.tasks.api import InstrumentTask, TaskInterface, validators
+from ecpy.utils.atom_util import ordered_dict_to_pref, ordered_dict_from_pref
+
+class SynthHDsetChannelInterface(TaskInterface):
+    """Set the central frequency to be used for the specified channel.
+
+    """
+    #: Id of the channel whose central frequency should be set.
+    channel = Int(0).tag(pref=True)
+
+    def perform(self, frequency=None):
+        """Set the central frequency of the specified channel.
+
+        """
+        task = self.task
+        channel = self.channel
+        task.driver.channel = channel
+
+        task.i_perform()
+
+    def check(self, *args, **kwargs):
+        """Make sure the specified channel does exists on the instrument.
+
+        """
+        test, traceback = super(SynthHDsetChannelInterface, self).check(*args,
+                                                                **kwargs)
+        return test, traceback
+
```

Comment:
All the following classes do not belong here and are duplicated please remove them.